tdrs-frontend/src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useRef, useEffect } from 'react'
+import Button from '../Button'
+
+const Modal = ({ title, message, buttons = [], isVisible = false }) => {
+  const modalRef = useRef(null)
+  useEffect(
+    () =>
+      isVisible && modalRef && modalRef.current
+        ? modalRef.current
+            .querySelector(`button[buttonkey="${buttons[0].key}"]`)
+            .focus()
+        : undefined,
+    [isVisible, modalRef, buttons]
+  )
+
+  const onTabPressed = (shiftKey = false) => {
```

Comment:
along with the `focus()` in the `useEffect` above, this is what "traps" the keyboard focus into the modal. the user should only be able to tab through (forward or backward by adding a `shiftKey` modifier`) buttons within the modal's list.